k2/csrc/fsa_utils.cu

```diff
@@ -228,14 +236,20 @@ static Fsa K2TransducerFromStream(std::istringstream &is,
   K2_CHECK_EQ(finished, true) << "The last line should be the final state";
```

Comment:
(this relates to a few lines above)...
I don't think we should be ignoring the final_state, as in:
` (void)StringToInt(splits[0]);`.  Instead, I think it's better if the interface of  FsaFromArray1 is extended to
take the final-state.  Otherwise we may do something very unexpected if the input is invalid.

Reply:
.. this code is a bit of a mess because once I had the idea that the best way to represent FSAs was simply arrays of arcs, keeping the final-state out of it, but this turned out to be not really practical.